### ISSUE TITLE
Make `StatusListPicker` work with backend models

### ISSUE THREAD

**Issue by osmaczko** [status-im/status-desktop#6500]
### Description

`StatusListPicker` is a reusable StatusQ component which purpose is to display a list of items that comply with the predefined interface. It expects `ListModel` with elements that define `key, name, shortName, symbol, imageSource, category, selected`.
The interface is good enough for a sandbox app that works on mocked `ListModel`s, but it does work with status-desktop which provides data through `QAbstractListModel`.

Using `ListModel` as an interface leads to situations where `StatusListPicker` drives the backend model structure, which is **entirely wrong**. 
Example: 
[LanguageStore.qml](https://github.com/status-im/status-desktop/blob/ee8477ae9b588a98213427cdbefe2dc9ab1ed3c3/ui/app/AppLayouts/Profile/stores/LanguageStore.qml#L13-L29)
```
    property ListModel languageModel: ListModel {
        ListElement {key: "en"; shortName: "English"; name: "English"; category: ""; imageSource: "../../assets/twemoji/svg/1f1ec-1f1e7.svg"; selected: false}
        ListElement {key: "zh"; shortName: "普通话"; name: "Chinese (Mainland China)"; imageSource: "../../assets/twemoji/svg/1f1e8-1f1f3.svg"; category: ""; selected: false}
        ListElement {key: "ko"; shortName: "한국어"; name: "Korean"; category: ""; imageSource: "../../assets/twemoji/svg/1f1f0-1f1f7.svg"; selected: false}
        ListElement {key: "pt_BR"; shortName: "Português"; name: "Portuguese (Brazil)"; category: ""; imageSource: "../../assets/twemoji/svg/1f1e7-1f1f7.svg"; selected: false}
        ListElement {key: "ru"; shortName: "Русский Язык"; name: "Russian"; category: ""; imageSource: "../../assets/twemoji/svg/1f1f7-1f1fa.svg"; selected: false}
        ListElement {key: "ar"; shortName: "اَلْعَرَبِيَّةُ"; name: "Arabic"; category: qsTr("Beta Languages"); imageSource: "../../assets/twemoji/svg/1f1f8-1f1e6.svg"; selected: false}
        ListElement {key: "zh_TW"; shortName: "臺灣華語"; name: "Chinese (Taiwan)"; category: qsTr("Beta Languages"); imageSource: "../../assets/twemoji/svg/1f1f9-1f1fc.svg"; selected: false}
        ListElement {key: "de"; shortName: "Nederlands"; name: "Dutch"; category: qsTr("Beta Languages"); imageSource: "../../assets/twemoji/svg/1f1f3-1f1f1.svg"; selected: false}
        ListElement {key: "fil"; shortName: "Wikang Filipino"; name: "Filipino"; category: qsTr("Beta Languages"); imageSource: "../../assets/twemoji/svg/1f1f5-1f1ed.svg"; selected: false}
        ListElement {key: "fr"; shortName: "Français"; name: "French"; category: qsTr("Beta Languages"); imageSource: "../../assets/twemoji/svg/1f1eb-1f1f7.svg"; selected: false}
        ListElement {key: "id"; shortName: "Bahasa Indonesia"; name: "Indonesian"; category: qsTr("Beta Languages"); imageSource: "../../assets/twemoji/svg/1f1ee-1f1e9.svg"; selected: false}
        ListElement {key: "it"; shortName: "Italiano"; name: "Italian"; category: qsTr("Beta Languages"); imageSource: "../../assets/twemoji/svg/1f1ee-1f1f9.svg"; selected: false}
        ListElement {key: "es"; shortName: "Español"; name: "Spanish"; category: qsTr("Beta Languages"); imageSource: "../../assets/twemoji/svg/1f1ea-1f1f8.svg"; selected: false}
        ListElement {key: "tr"; shortName: "Türkçe"; name: "Turkish"; category: qsTr("Beta Languages"); imageSource: "../../assets/twemoji/svg/1f1f9-1f1f7.svg"; selected: false}
        ListElement {key: "ur"; shortName: "اُردُو"; name: "Urdu"; category: qsTr("Beta Languages"); imageSource: "../../assets/twemoji/svg/1f1f5-1f1f0.svg"; selected: false}
    }
```
In order to make a language model work with `StatusListPicker` its fields were renamed, as a consequence making this model incompatible with any other StatusQ component that may require different list elements naming (e.g. `descr` instead of `shortName`, etc.). _Mentioned limitation makes it impossible to use `StatusListPicker` with models with different role names._

In short: `StatusListPicker` should be able to work with different kinds of models, for instance, we may want to "pick" user, language, currency, etc. (these models have different role names).

Solution proposal:
- extract delegate so that each client assigns properties accordingly to model roles:
```
StatusListPicker {
    model: userModel
    delegate: StatusListPickerDelegate {
        key: model.name
        shortName: model.alias
        etc..
    }
}
```

Obstacles:
- `StatusListPicker` creates `filteredModel` imperatively assuming role names. It has to be abstracted as well. That may be the challenging part.

**Comment by caybro** [comment]
QAbstractListModel to the rescue :)
